modules/external_petsc_solver/src/mesh/PETScDMDAMesh.C

```diff
@@ -186,7 +186,8 @@ add_element_Quad4(DM da,
   Elem * elem = new Quad4;
   elem->set_id(elem_id);
   elem->processor_id() = pid;
-  elem->set_unique_id() = elem_id;
+  // Make sure our unique_id doesn't overlap any nodes'
+  elem->set_unique_id() = elem_id + (nx + 1) * (ny + 1);
```

Comment:
So unique ID will be different from ID?

Reply:
Yes.  That's pretty common, e.g. when we reuse id values during AMR/C but never unique_id values.  It's never safe to assume the contrary.